onebusaway-android/src/main/java/org/onebusaway/android/ui/TripDetailsListFragment.java

```diff
@@ -704,4 +689,65 @@ public Integer getNextStopIndex() {
             return mNextStopIndex;
         }
     }
+
+    private void setScroller(final ListView listView)
+    {
+        if (mScrollMode == null) {
+            // Don't automatically scroll to stop or vehicle icon
+            return;
+        }
+
+        switch (mScrollMode) {
+            case SCROLL_MODE_VEHICLE: {
+                // Scroll to bus marker if scroll mode is set to vehicle & we have index.
+                final Integer nextStop = mAdapter.getNextStopIndex();
+                if (nextStop != null) {
+                    listView.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            listView.setSelection(nextStop - 1);
+                        }
+                    });
+                } else {
+                    // If we couldn't get the bus marker, fall back to stop.
+                    mStopIndex = findIndexForStop(mTripInfo.getSchedule().getStopTimes(), mStopId);
+                    if (mStopIndex != null) {
+                        listView.post(new Runnable() {
+                            @Override
+                            public void run() {
+                                listView.setSelection(mStopIndex);
+                            }
+                        });
+                    }
+                }
+            }
```

Comment:
I think you're missing a `break` here - testing the PR w/ breakpoints it seems to execute both `case` blocks.